src/core/lib/iomgr/tcp_client_cfstream.cc

```diff
@@ -140,10 +140,12 @@ static void OnOpen(void* arg, grpc_error_handle error) {
 
 static void ParseResolvedAddress(const grpc_resolved_address* addr,
                                  CFStringRef* host, int* port) {
-  std::string host_port = grpc_sockaddr_to_string(addr, true);
+  auto host_port = grpc_sockaddr_to_string(addr, true);
   std::string host_string;
   std::string port_string;
-  grpc_core::SplitHostPort(host_port, &host_string, &port_string);
+  grpc_core::SplitHostPort(
+      host_port.ok() ? host_port.value() : host_port.status().ToString(),
```

Comment:
It does not make sense to call `SplitHostPort()` on the error message string.  Let's just crash here.

Reply:
Done.